nym-wallet/src/pages/bonding/node-settings/node-settings.constant.tsx

```diff
@@ -0,0 +1,7 @@
+enum Options {
+  General,
+  Unbond,
+}
+type NavItem = keyof typeof Options;
+
+export const nodeSettingsNav: NavItem[] = ['General', 'Unbond'];
```

Comment:
Instead of using enum (which are JS thing), I always recommend to use TS types instead.

So here a more "TS" way to do this is using its typing system again :p
```
export const navItems = ['General', 'Unbond'] as const;

export type NodeSettingsNav = typeof navItems[number]; // type NodeSettingsNav = 'General' | 'Unbond';
```
With that, instead of having some runtime values, you have a real type you can rely on
```type NodeSettingsNav = 'General' | 'Unbond';```

Okay now we need to do a bit of refactoring to embrace it

First you need to refactor a bit `nym-wallet/src/components/Tabs.tsx`, change the signature of the component to accept a `readonly` array of string (this is what we want as we expect tabs to stay the same). Plus, we need to change the type of the current selected tab to the value of the tab itself and not an index (which can be unreliable):
```
type Props = {
  tabs: readonly string[];
  selectedTab: string;
  disabled?: boolean;
  onChange?: (event: React.SyntheticEvent, tab: string) => void;
  disableActiveTabHighlight?: boolean;
  tabSx?: SxProps;
  tabIndicatorStyles?: {};
};

export const Tabs = ({
  tabs,
  selectedTab,
  disabled,
  disableActiveTabHighlight,
  onChange,
  tabSx,
  tabIndicatorStyles,
}: Props) => (
```
Note that I voluntarily dropped the use of `React.FC` because it's bad to use it (why? check [this](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/))

Dont forget to map the corresponding value for each Tab:
```
      <Tab key={tabName} label={tabName} sx={{ textTransform: 'capitalize' }} value={tabName} disabled={disabled} />
```

Now, you can simplify the logic in `nym-wallet/src/pages/bonding/node-settings/NodeSettings.tsx`
```
  const [value, setValue] = React.useState<NodeSettingsNav>('General');

  const handleChange = (event: React.SyntheticEvent, tab: string) => {
    setValue(tab as NodeSettingsNav);
  };

  useEffect(() => {
    if (location.state === 'unbond') {
      setValue('Unbond');
    }
  }, [location]);

// ...
              <Tabs
                tabs={navItems}
// ...
        <Divider />
        {value === 'General' && bondedNode && <NodeGeneralSettings bondedNode={bondedNode} />}
        {value === 'Unbond' && bondedNode && (

```
As you can see, the code is a bit more clean, but most importantly, our state is typed to be a union type of `General` or `Unbond`. So you can't mess up with wrong mapping between value and index, for example.

Finally, don't forget to refactor as well `nym-wallet/src/components/Bonding/modals/NodeSettingsModal.tsx`
```
      <Tabs tabs={['System variables']} selectedTab="System variables" disableActiveTabHighlight />
```

That's it :)

Please all my above note are only suggestion, the code you wrote works well and is perfectly functional. I just try to share a more "idiomatic" way to work with TS :) Which lead to cleaner code base in the long run.